docs/user/operator-hub.adoc

```diff
@@ -0,0 +1,253 @@
+:source-highlighter: pygments
+
+# odo integration with Operator Hub
+---
+
+When working in experimental mode, Odo provides the ability to work with
+link:https://www.openshift.com/learn/topics/operators[Operators] installed on
+the cluster. It allows listing of Operators, creation of services from CRD
+provided by the Operators, printing the YAML definition and providing custom
+YAML definition to start the service from a CRD. 
+
+[NOTE]
+====
+Installation of Operators is not a part of odo workflow. It is something that
+your OpenShift/Kubernetes cluster administrator should be able to do for you. 
+====
+
+In this document we will cover the following:
+
+. <<list-operators,List the Operators installed in the current project>>
+. <<dry-run,Print the YAML specification that will be used to start the service from a CRD>>
+. <<create-service,Create a service from the Custom Resource Definitions (CRD) provided by the Operators>>
+. <<create-service-from-yaml,Create a service from the YAML stored in a file on local system>>
+
+[NOTE]
+====
+For every command mentioned below to work as expected, we first need to enable
+the experimental mode. This can be done in different ways. Make sure that you
+have installed latest version of odo and perform any one of the following actions.
+
+* Enable experimental mode in odo preferences:
++
+[source,shell]
+----
+$ odo prefrence set Experimental true
+----
+
+
+* Export the environment variable `ODO_EXPERIMENTAL`:
++
+[source,shell]
+----
+$ export ODO_EXPERIMENTAL=true
+----
+
+* Prefix every command mentioned below with `ODO_EXPERIMENTAL=true`. For
+  example, to list the Operators installed in current project:
++
+[source,shell]
+----
+$ ODO_EXPERIMENTAL=true odo catalog list services
+----
+====
+
+=== [[list-operators]]Listing the Operators
+
+To list the Operators installed in current project, execute below command:
+
+[source,shell]
+----
+$ odo catalog list services
+----
+
+It will list the
+link:https://docs.openshift.com/container-platform/4.3/operators/olm-what-operators-are.html[Operators]
+and the
+link:https://docs.openshift.com/container-platform/4.3/operators/crds/crd-extending-api-with-crds.html#crd-custom-resource-definitions_crd-extending-api-with-crds[CRD
+(Custom Resource Definitions)] provided by these Operators. For example, we
+have installed Etcd and MongoDB Operators and the output we get is like below:
+
+[source,shell]
+----
+$ odo catalog list services                       
+Operators available in the cluster
+NAME                          CRDs
+etcdoperator.v0.9.4           EtcdCluster, EtcdBackup, EtcdRestore
+mongodb-enterprise.v1.4.5     MongoDB, MongoDBUser, MongoDBOpsManager
+----
+
+In above output, `etcdoperator.v0.9.4` is the operator while `EtcdCluster`,
+`EtcdBackup` and `EtcdRestore` are the CRDs provided by this operator.
+
+When starting a service, we will be doing that from one of the CRDs of an
+operator
+
```

Comment:
nit: To start a service from operator we need `operator  name` and a `crd`

Reply:
@adisky IMO, we've detailed that info all over the place in the doc. But if you insist, I don't mind adding it here. :slightly_smiling_face:

Reply:
@dharmit I was not saying to add it, replace this `When starting a service, we will be doing that from one of the CRDs of an
operator ` with ` we need operator name and a crd to start a service from operator`. IMO The above line is confusing to read.

Reply:
Done.